File: app/src/main/java/com/example/myapplication/ViewModels/StoryViewModel.java
package com.example.myapplication.ViewModels;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.myapplication.Models.Story;
import com.example.myapplication.Repositories.StoryRepository;

import java.util.List;

public class StoryViewModel extends AndroidViewModel {

    private StoryRepository repository;
    private LiveData<List<Story>> allStories;

    public StoryViewModel(@NonNull Application application) {
        super(application);

        repository = new StoryRepository(application);
        allStories = repository.getAllStories();
    }

    public void insert(Story story) {
        repository.insert(story);
    }

    public void update(Story story) {
        repository.update(story);
    }

    public void delete(Story story) {
        repository.delete(story);
    }

    public void deleteAllStories() {
        repository.deleteAllStories();
    }

    public LiveData<List<Story>> getAllStories() {
        return allStories;
    }

}

File: app/src/main/java/com/example/myapplication/ViewModels/StoryViewModel.kt
package com.example.myapplication.ViewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import com.example.myapplication.Models.Story
import com.example.myapplication.Repositories.StoryRepository

class StoryViewModel(application: Application) : AndroidViewModel(application) {
    private val repository: StoryRepository
    val allStories: LiveData<List<Story>>
    fun insert(story: Story?) {
        repository.insert(story)
    }

    fun update(story: Story?) {
        repository.update(story)
    }

    fun delete(story: Story?) {
        repository.delete(story)
    }

    fun deleteAllStories() {
        repository.deleteAllStories()
    }

    init {
        repository = StoryRepository(application)
        allStories = repository.allStories
    }
}